erc721-tests/contracts/Erc-721.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "hardhat/console.sol";
+
+contract Erc721 is Ownable, ERC721("SolicyNFT", "HNFT") {
+    uint256 tokenId = 0;
+    string public _name;
+    string public _symbol;
+    string public __baseURI;
+    string public uriSuffix = ".json";
+    mapping(address => TokenMetaData[]) public ownershipRecord;
+    mapping(uint256 => bool) internal lockRecord;
+
+    struct TokenMetaData {
+        string name;
+        uint tokenId;
+        uint256 lockedTimeStamp;
+        uint256 daysToLock;
+        string tokenURI;
+        uint experience;
+        uint rank;
+    }
+
+    modifier isNotLocked (uint256 tokenId_) {
+        require(lockRecord[tokenId_] == false, "The token is loked");
+        _;
+    }
+
+    constructor(string memory name_, string memory symbol_, string memory baseURI_) {
+        _name = name_;
+        _symbol = symbol_;
+        __baseURI = baseURI_;
+    }
+
+    function name() public view virtual override returns (string memory) {
+        return _name;
+    }
+
+    function symbol() public view virtual override returns (string memory) {
+        return _symbol;
+    }
+
+    function _baseURI() internal view virtual override returns (string memory) {
+        return __baseURI;
+    }
+
+    function tokenURI(uint256 tokenId_) public view virtual override returns (string memory) {
+        _requireMinted(tokenId_);
+
+        return bytes(_baseURI()).length > 0 ? string(abi.encodePacked(_baseURI(), Strings.toString(tokenId_), uriSuffix)) : "";
+    }
+
+    function mintToken(address recipient, string memory name_) onlyOwner public {
+        require(owner()!=recipient, "Recipient cannot be the owner of the contract");
+        _safeMint(recipient, tokenId);
+        ownershipRecord[recipient].push(TokenMetaData(name_, tokenId, 0, 0, tokenURI(tokenId), 0, 0));
+        tokenId = tokenId + 1;
+    }
+
+    function transfer(
+        address from_,
+        address to_,
+        uint256 tokenId_
+    ) external virtual isNotLocked(tokenId) {
+        _transfer(from_, to_, tokenId_);
+    }
+
+    function burn(uint256 tokenIdToBurn, int256 tokenIdToTransferExp) external virtual isNotLocked(tokenId) {
+        require(_ownerOf(tokenIdToBurn) != address(0), "ERC721: invalid token ID");
+        require(tokenIdToTransferExp > 0, "ERC721: invalid reciver token ID");
+
+        uint senderexp = (getTockenMetadata(tokenIdToBurn).experience / 100) * 80;
```

Comment:
Multiply first then divide. If experience is 99, for example, the senderexp will be 0.